src/UKHO.ERPFacade.Common/IO/Azure/AzureTableReaderWriter.cs

```diff
@@ -379,6 +405,35 @@ await foreach (var entity in entities)
             return records.FirstOrDefault();
         }
 
+        public async Task UpdateStatusOfRecordOfSaleEntityPublisedEvent(string correlationId)
+        {
+            TableClient tableClient = GetTableClient(RecordOfSaleTableName);
+            RecordOfSaleEventEntity existingEntity = await GetRecordOfSaleEventEntity(correlationId);
+
+            if (existingEntity != null)
+            {
+                existingEntity.Status = "Complete";
+                await tableClient.UpdateEntityAsync(existingEntity, ETag.All, TableUpdateMode.Replace);
+
+                _logger.LogInformation(EventIds.UpdatedStatusOfRecordOfSalePublishedEventInAzureTable.ToEventId(), "Status of existing record of sale published event updated in azure table successfully.");
+            }
+        }
+
+        public async Task UpdateStatusOfUploadedLicenceRecordOfSaleEntityEvent(string correlationId)
+        {
+            TableClient tableClient = GetTableClient(LicenceUpdateTableName);
+            RecordOfSaleEventEntity existingEntity = await GetRecordOfSaleLicenceEventEntity(correlationId);
+
+            if (existingEntity != null)
+            {
+                existingEntity.Status = "Complete";
+                await tableClient.UpdateEntityAsync(existingEntity, ETag.All, TableUpdateMode.Replace);
+
+                _logger.LogInformation(EventIds.UpdatedStatusOfUploadedLicenceRecordOfSaleEntityEvent.ToEventId(), "Existing Record Of Sale published event is updated in azure table successfully.");
```

Comment:
Please include Licence Updated event name in message description

Reply:
Message updated as "Status of existing licence updated published event updated in azure table successfully".